apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/tag/AbstractTag.java

```diff
@@ -60,4 +61,21 @@ public int getId() {
     public boolean isCanOverwrite() {
         return canOverwrite;
     }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o)
+            return true;
+        if (!(o instanceof AbstractTag))
+            return false;
+        final AbstractTag<?> that = (AbstractTag<?>) o;
+        return getId() == that.getId() &&
+            isCanOverwrite() == that.isCanOverwrite() &&
+            key.equals(that.key);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getId(), isCanOverwrite(), key);
+    }
```

Comment:
What are these for?

Reply:
For stringTag equals:
![image](https://user-images.githubusercontent.com/31562192/96411822-af560800-121b-11eb-8bbc-5386ac448fa8.png)